packages/vscode-extension/src/commonlib/azureLogin.ts

```diff
@@ -370,10 +389,24 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
     }
   }
 
-  // TODO add login and select subscription logic later
-  getSelectedSubscription(triggerUI=false): Promise<SubscriptionInfo | undefined> {
+  async getSelectedSubscription(triggerUI = false): Promise<SubscriptionInfo | undefined> {
     const azureAccount = this.getAzureAccount();
-    if (azureAccount.status === loggedIn) {
+    if (triggerUI) {
+      if (azureAccount.status != loggedIn) {
+        await this.login(true);
+      }
+      if (azureAccount.status === loggedIn && !AzureAccountManager.subscriptionId) {
+        await this.selectSubscription();
+      }
+    }
+    // no need to select sub when user only have one
+    if (azureAccount.status === loggedIn && !AzureAccountManager.subscriptionId && !triggerUI) {
```

Comment:
what's `!triggerUI` for? Could this single subs check always be there?
Or, move this part to `else` (if (triggerUI) ... **else ...**)

Reply:
Fixed.